### ISSUE TITLE
Support geoTIFF

### ISSUE THREAD

**Issue by sonora** [osmandapp/OsmAnd#16373]
As a follow-up to  #9944, and building on the work already implemented here https://github.com/osmandapp/OsmAnd/commit/6c680cd81f4f1627b8cd252976ef064c6509a340, can I please log the feature request here to support the geoTIFF format as OsmAnd's main, overlay, and underlay map source.

**Comment by vshcherb** [comment]
That's not really planned for now we will need more specific examples probalby why could it be useful. As for now you can export GeoTiff as tiles and use it directly. Also geotiff only supported now as heightmap and only with mercator projection

**Comment by sonora** [comment]
The increasingly significant use case is that several high precision sources for scientific purposes and even some public administrations these days provide geoTIFF as their standard output.

As an example see https://geodaten.bayern.de/opengeodata/OpenDataDetail.html?pn=dop40 for the German state of Bavaria, where you can download high-resolution squares only as geoTIFF.

**Comment by vshcherb** [comment]
Will it support it in a reasonable manner - GB tif or we will need to preprocess it. This obviously requires investigation probably we can add this year experimental support to do some investigation but I doubt on mobile it will work flawlessly.

**Comment by ahgera** [comment]
Another use case for georeferenced raster files is map signs at trail information signs, these often contain information like trail color, shelters, etc. I know there are [apps](https://play.google.com/store/apps/details?id=com.custommapsapp.android) that lets you take a photo of such a map, set the position of a few points on it, and produce a georeferenced raster image as a result, which can be very handy when you need an ad hoc map.
![image](https://user-images.githubusercontent.com/34133296/216436002-1441a4ea-bc1a-4db0-b8fc-265bab326510.png)

Being able to georeference an image directly in OsmAnd, maybe using a plugin, and then loading it as an overlay layer would be really handy, but limiting the scope to support for various already georeferenced raster images would probably be better.

Depending on how you structure the code internally this could also be used to [improve kml/kmz support](https://github.com/osmandapp/OsmAnd/issues/11834), since those can contain image overlays as well.

**Comment by sonora** [comment]
Yes, true, but that use case talks about _creating_ a geo-aligned picture file, I was talking about  just _viewing_ a (geoTIFF  standard) one in OsmAnd.

**Comment by akkana** [comment]
I stumbled on this issue as I was looking for something else, and thought it might be worth commenting that you can convert geotiff to OsmAnd sqlite pretty easily with the help of gdal plus an extra script. I detailed the steps here: https://shallowsky.com/blog/mapping/osmand-making-overlay-maps.html (scroll down to "Convert the GeoTiff to Map Tiles"). I used that to make a land ownership OsmAnd overlay that I use every time I hike.

**Comment by Mis012** [comment]
@vshcherb
>we will need more specific examples probalby why could it be useful.

So I was wondering what the big deal is when you can just convert the geotiff to mbtiles, until I tried to do it and realized that if your geotiff doesn't happen to have the exact resolution of some mbtiles "zoom level", you will either lose resolution or end up with a much larger file than should be necessary (not sure how much larger because I ran out of disk space at 35GiB from 23GiB of geotiffs (would likely be even less after merging them))

TL;DR mbtiles don't support arbitrary resolution, and at least in some cases this is very problematic

Requiring Mercator projection would not be the worst thing ever, mbtiles already require that anyway, although even with geotiffs it's obviously not ideal and there is a reason other projections exist.

**Comment by Mis012** [comment]
hm, so I tried to use JPEG, the resulting file is ~9GiB, and neither QGIS (before converting to OsmAnd sqlite) nor OsmAnd (after converting) want to load it...
